core/controllers/creator_dashboard_test.py

```diff
@@ -644,6 +674,44 @@ def test_can_see_subscribers(self):
         response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
         self.assertEqual(len(response['subscribers_list']), 0)
 
+    def test_get_response_when_new_structure_players_is_true(self):
+        self.login(self.OWNER_EMAIL)
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.get_json(feconf.CREATOR_DASHBOARD_DATA_URL)
+            self.assertEqual(len(response['topic_summary_dicts']), 0)
+        self.logout()
+
+    def test_can_update_display_preference(self):
+        self.login(self.OWNER_EMAIL)
+        response_display_preference = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['display_preference']
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.post_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL,
+            {'display_preference': response_display_preference},
+            csrf_token=csrf_token)
+        self.logout()
+
+    def test_can_create_new_collection(self):
+        self.set_admins([self.OWNER_USERNAME])
+        self.login(self.OWNER_EMAIL)
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.post_json(feconf.NEW_COLLECTION_URL, {}, csrf_token=csrf_token)
+        self.logout()
+
+    def test_collections_list_gets_updated(self):
+        self.set_admins([self.OWNER_USERNAME])
+        self.login(self.OWNER_EMAIL)
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.post_json(feconf.NEW_COLLECTION_URL, {}, csrf_token=csrf_token)
+        response_collection_list = self.get_json(
+            feconf.CREATOR_DASHBOARD_DATA_URL)['collections_list']
+        self.assertNotEqual(response_collection_list, [])
```

Comment:
Same thing. Testing the empty case is not a bad idea, but you should also test the "typical" case as a minimum baseline.

I'm going to stop here -- please take a pass on the tests and re-request review once you've made them more robust. Thanks!

Reply:
done